source/server/session_resource_repository.h

```diff
@@ -145,17 +145,24 @@ TResourceHandle SessionResourceRepository<TResourceHandle>::access_session(uint3
   return resource_map_.get_handle(id);
 }
 
+template <typename TResourceHandle>
+uint32_t SessionResourceRepository<TResourceHandle>::access_session_id(uint32_t session_id, const std::string& session_name) const
+{
+  return session_repository_->access_session(session_id, session_name);
+}
+
 template <typename TResourceHandle>
 uint32_t SessionResourceRepository<TResourceHandle>::resolve_session_id(TResourceHandle handle) const
```

Comment:
This is still needed by some drivers, hence the `reverse_map_` sticking around.

Basically, any service that has a method where the driver api outputs a session type (like `NiRFSGService::GetAttributeViSession`) and we want to convert that back to our grpc `Session` type before returning. For those drivers where the one to one mapping between instrument handle and grpc Session holds this should be fine.

I'm assuming that RFmx will not need this. I checked the headers and there are no outputs of session type except the initialize methods which are handled separately and don't go through this. RFmx's `reverse_map_`. And the cross driver session methods should now use the newly added `access_session_id` instead of `resolve_session_id`.

Reply:
TODO: I think I'll try to add something to fail generation if we try to add a call to `resolve_session_id` when generating an RFmx service. I'll probably add it to the mako for the service.cpp in the one spot where it adds the `resolve_session_id` call.

Reply:
Would that be something in config.py like "disallow_session_reverse_lookup" or something?

Reply:
We can double check with Nagashree if there's any reverse lookup case we might need to support. She's supposed to put together a list of scenarios for us.